src/System.IO.FileSystem/src/System/IO/WinRTFileSystem.cs

```diff
@@ -227,14 +228,121 @@ public override bool DirectoryExists(string fullPath)
 
         public override IEnumerable<string> EnumeratePaths(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new string[0];
+            IReadOnlyList<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
+            return TranslatePaths(storageFiles);
         }
 
         public override IEnumerable<FileSystemInfo> EnumerateFileSystemInfos(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new FileSystemInfo[0];
+            IReadOnlyList<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
+            return TranslateFileSystemInfos(storageFiles);
+        }
+
+        /// <summary>
+        /// Translates IStorageItems into FileSystemInfos
+        /// </summary>
+        private static IEnumerable<FileSystemInfo> TranslateFileSystemInfos(IReadOnlyList<IStorageItem> storageFiles)
+        {
+            // Translate the StorageFile results into FileSystemInfos
+            int count = storageFiles.Count;
+            for (int i = 0; i < count; i++)
+            {
+                if (storageFiles[i].IsOfType(StorageItemTypes.Folder))
+                {
+                    yield return new DirectoryInfo(storageFiles[i].Path);
+                }
+                else // If it is neither a File nor folder then we treat it as a File.
+                {
+                    yield return new FileInfo(storageFiles[i].Path);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Translates IStorageItems into string paths
+        /// </summary>
+        private static IEnumerable<string> TranslatePaths(IReadOnlyList<IStorageItem> storageFiles)
+        {
+            // Yield the paths of the StorageItems
+            int count = storageFiles.Count;
+            for (int i = 0; i < count; i++)
+            {
+                yield return storageFiles[i].Path;
+            }
+        }
+
+        private async static Task<IReadOnlyList<IStorageItem>> EnumerateFileQuery(string path, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
+        {
+            // Get a StorageFolder for "path"
+            string fullPath = Path.GetFullPath(path);
+            StorageFolder folder;
+            try
+            {
+                folder = await StorageFolder.GetFolderFromPathAsync(fullPath).TranslateWinRTTask(fullPath, isDirectory: true);
```

Comment:
[TranslateWinRTTask returns the `ConfiguredTaskAwaitable` object directly](https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/src/System/IO/WinRTIOExtensions.cs). If we have to ConfigureAwait on the result of that anyways, then the helper method has no purpose and we should be calling `TranslateWinRTTaskCore` from WinRTFileSystem. I'm guessing the Translate method was added as a helper to avoid the need for a ConfigureAwait at every WinRT async call site; if it's not doing that then it should be removed to save us the indirection.